core/src/main/scala/akka/kafka/internal/KafkaConsumerActor.scala

```diff
@@ -340,72 +339,15 @@ class KafkaConsumerActor[K, V](settings: ConsumerSettings[K, V]) extends Actor w
-      )
-      if (settings.wakeupDebug && wakeups > settings.maxWakeups / 2) {
-        val stacks =
-          Thread.getAllStackTraces.asScala.map { case (k, v) => s"$k\n ${v.mkString("\n")}" }.mkString("\n\n")
-        log.warning(
-          "Wake up has been triggered {} times (See setting akka.kafka.consumer.wakeup-debug). Dumping stacks: {}",
-          wakeups,
-          stacks
-        )
-      }
-      consumer.wakeup()
-    }(context.system.dispatcher)
+
     val currentAssignmentsJava = consumer.assignment()
 
-    def tryPoll(timeout: Long): ConsumerRecords[K, V] =
+    def tryPoll(timeout: FiniteDuration): ConsumerRecords[K, V] =
       try {
-        val records = consumer.poll(timeout)
+        val records = consumer.poll(timeout.asJava)
         initialPoll = false
-        wakeups = 0
         records
       } catch {
-        case w: WakeupException =>
-          wakeups = wakeups + 1
-          if (wakeups == settings.maxWakeups) {
-            log.error(
-              "WakeupException limit exceeded during poll({}), stopping (max-wakeups = {}, wakeup-timeout = {}).",
-              timeout,
-              settings.maxWakeups,
-              settings.wakeupTimeout.toCoarsest
-            )
-            context.stop(self)
-          } else {
-            if (log.isWarningEnabled && wakeups > settings.maxWakeups / 2) {
-              log.warning(
-                "Consumer poll({}) interrupted with WakeupException (#{} of max-wakeups = {}, wakeup-timeout = {}).",
-                timeout,
-                wakeups,
-                settings.maxWakeups,
-                settings.wakeupTimeout.toCoarsest
-              )
-              if (initialPoll) {
-                log.error(
-                  "Initial consumer poll({}) with bootstrap servers {} did not succeed, still trying",
-                  timeout,
-                  settings.properties.getOrElse(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, "not set")
-                )
-              }
-            }
-
-            // If the current consumer is using group assignment (i.e. subscriptions is non empty) the wakeup might
-            // have prevented the re-balance callbacks to be called leaving the Source in an inconsistent state w.r.t
-            // assigned TopicPartitions. In order to reconcile the state we manually call callbacks for all added/remove
-            // TopicPartition assignments aligning the Source's state the consumer's.
-            // We are safe to perform the operation here since the poll() thread has been aborted by the wakeup call
-            // and there are no other threads are using the consumer.
-            // Note: in case of manual partition assignment this is not needed since rebalance doesn't take place.
-            if (subscriptions.nonEmpty) {
-              val newAssignments = consumer.assignment().asScala
-              reconcileAssignments(currentAssignmentsJava.asScala.toSet, newAssignments.toSet)
-            }
```

Comment:
wonder how this is handled now, if `poll` returns without any records?

perhaps it's handled by the metadata calls that are part of the poll?

Reply:
Looking at the code, that's what the case seems to be